client/js/table-view.js

```diff
@@ -86,26 +165,229 @@ class TableView {
 		this.sheetBodyEl.appendChild(fragment);
 	}
 
+	isCurrentCell(col, row) {
+		return this.currentCellLocation.col === col &&
+		       this.currentCellLocation.row === row;
+	}
+
 	attachEventHandlers() {
+
+		const context = this;
+
 		this.sheetBodyEl.addEventListener('click', this.
 			handleSheetClick.bind(this));
 		this.formulaBarEl.addEventListener('keyup', this.
 			handleFormulaBarChange.bind(this));
+		
+		this.addRowEl.addEventListener('click', this.
+			handleAddRowClick.bind(this));
+		this.addColumnEl.addEventListener('click', this.
+			handleAddColumnClick.bind(this));
+
+		this.rowLabelsEl.addEventListener('click', this.
+			handleRowLabelClick.bind(this));
+
+		this.headerRowEl.addEventListener('click', this.
+			handleColHeaderClick.bind(this));
+
+		// this works fine
+		this.computeSumFormulaEl.addEventListener('click', this.
+			handleFormulaBarEnter.bind(this));
+
+		// this doesn't work...
+		/*this.formulaBarEl.addEventListener('keypress', function(e) {
+			if (e.key === 'Enter') {
+				console.log('hi');
+				console.log(context.handleFormulaBarEnter);
+				console.log(context)
+				context.handleFormulaBarEnter.bind(e);
+			}
+		});*/
+			
+	}
+
+	handleFormulaBarEnter(event) {
+		const value = this.formulaBarEl.value;
+		
+		// to potentially use sum formula...
+		// must start with correct command, include colon, end with closing paren
+		if (value.substring(0, 5) === "=SUM(" && 
```

Comment:
This raw regular expression stuff is impressive that you got working... but also pretty raw.

Consider using a parser generator such as PEG JS instead.

Try integrating this PEG JS code (see code snippet below) which I have written for you. If you read through it you will find it's pretty intuitive to see how this parser definition specifies how to "consume" different parts of the input string.  The output is a clean object.
![img](http://i.imgur.com/zUPCgsj.png)

https://pegjs.org/online

```
sheet_function
  = _ equal_sign _ func:function_name _ params:params {
  	return {
    	func: func,
        params: params
    };
  }
  
equal_sign
  = '='
  
function_name
  = chars:[A-Z]* { return chars.join(''); }

params
  = '(' param:param* ')' {
  	return param;
  }

param
  = chars:[a-zA-Z0-9]+ _ delimiter? _ { return chars.join(''); }

delimiter
  = [:,]

_ // whitespace 
  = [ \t\n\r]*
```